File: src/main/java/net/hotamachisubaru/casino/Listener/CasinoListener.java
package net.hotamachisubaru.casino.Listener;

import net.hotamachisubaru.casino.Slot.SlotMachine;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.inventory.InventoryClickEvent;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.ItemStack;

public class CasinoListener implements Listener {

    private static final String CASINO_GAME_TITLE = "カジノゲーム選択";

    @EventHandler
    public void onInventoryClick(InventoryClickEvent event) {
        if (!(event.getWhoClicked() instanceof Player)) return;

        Player player = (Player) event.getWhoClicked();
        Inventory inventory = event.getClickedInventory();

        if (inventory == null || !event.getView().getTitle().equals(CASINO_GAME_TITLE)) return;
        event.setCancelled(true);

        ItemStack item = event.getCurrentItem();
        if (item == null || item.getItemMeta() == null) return;

        String itemName = item.getItemMeta().getDisplayName();
        if (itemName == null) return;

        handleItemClick(player, itemName, inventory);
    }

    private void handleItemClick(Player player, String itemName, Inventory inventory) {
        switch (itemName) {
            case "スロットマシン":
                player.sendMessage("いくら賭けますか？");
                player.sendMessage("金額をチャットで入力してください。");
                SlotMachine.waitForBetAmount(player, betAmount -> SlotMachine.openSlotGUI(player, betAmount));
                break;
            case "ルーレット":
                player.performCommand("roulette");
                break;
            case "ブラックジャック":
                player.performCommand("blackjack");
                break;
            default:
                break;
        }
    }
}

File: src/main/java/net/hotamachisubaru/casino/Listener/CasinoListener.kt
package net.hotamachisubaru.casino.Listener

import net.hotamachisubaru.casino.Slot.SlotMachine
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.inventory.InventoryClickEvent

class CasinoListener : Listener {
    @EventHandler
    fun onInventoryClick(event: InventoryClickEvent) {
        if (event.getWhoClicked() !is Player) return
        val player = event.getWhoClicked() as Player
        val clickedInventory = event.getClickedInventory()

        if (clickedInventory == null || event.getView().getTitle() != "カジノゲーム選択") return

        event.setCancelled(true)

        val clickedItem = event.getCurrentItem()
        if (clickedItem == null || clickedItem.getItemMeta() == null) return

        val itemName = clickedItem.getItemMeta().getDisplayName()
        when (itemName) {
            "スロットマシン" -> SlotMachine.openSlotGUI(player)
            "ルーレット" -> player.performCommand("roulette")
            "ブラックジャック" -> player.performCommand("blackjack")
            else -> {}
        }
    }
}

